src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -1056,16 +1053,23 @@ bool ColorControlServer::moveToHueCommand(app::CommandHandler * commandObj, cons
         colorHueTransitionState->finalHue = static_cast<uint8_t>(hue);
     }
 
-    colorHueTransitionState->stepsRemaining = transitionTime;
-    colorHueTransitionState->stepsTotal     = transitionTime;
+    colorHueTransitionState->stepsRemaining = numberOfSteps;
+    colorHueTransitionState->stepsTotal     = numberOfSteps;
     colorHueTransitionState->endpoint       = endpoint;
     colorHueTransitionState->up             = (direction == HueDirection::kUp);
     colorHueTransitionState->repeat         = false;
 
-    SetHSVRemainingTime(endpoint);
+    if (transitionTime > 0)
+    {
+        SetHSVRemainingTime(endpoint);
+    }
+    else
+    {
+        Attributes::RemainingTime::Set(endpoint, 0);
```

Comment:
This doesn't consider if there is an active saturation. RemainingTime needs to be set to the greater of the two.

Same for the other command where you apply this

Reply:
I've removed the conditional here so it continues to call SetHSVRemainingTime. I've added another field to the transition state that tracks the time remaining.